Python/Product/VSInterpreters/InterpreterOptionsService.cs

```diff
@@ -589,5 +598,74 @@ IServiceProvider serviceProvider
             info._lock.Release();
             return res;
         }
+
+        private sealed class InterpretersEnumerator : IEnumerator<IPythonInterpreterFactory> {
+            private readonly InterpreterOptionsService _owner;
+            private readonly IEnumerator<IPythonInterpreterFactory> _e;
+
+            public InterpretersEnumerator(InterpreterOptionsService owner, IEnumerator<IPythonInterpreterFactory> e) {
+                _owner = owner;
+                _owner.BeginSuppressInterpretersChangedEvent();
+                _e = e;
+            }
+
+            public void Dispose() {
+                Dispose(true);
+                GC.SuppressFinalize(this);
+            }
+
+            private void Dispose(bool disposing) {
+                if (disposing) {
+                    _e.Dispose();
+                }
+                _owner.EndSuppressInterpretersChangedEvent();
+            }
+
+            ~InterpretersEnumerator() {
+                Debug.Fail("Interpreter enumerator should always be disposed");
+                Dispose(false);
+            }
+
+            public IPythonInterpreterFactory Current { get { return _e.Current; } }

```

Comment:
I think we only put methods all on a single line when the body is `{}`. And even then there are blank lines to separate them to make it clear they are different.